### ISSUE TITLE
Fix pyflakes for words

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6994]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#6994|
|Type|enhancement|
|Created|2014-02-27 12:36:46Z|
|Branch|https://github.com/twisted/twisted/tree/pyflakes-errors-twisted-words-6994|

Attachments:

* [6994-1.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/47a/47af2040bb2012893930fef26cd9f64525453d04/69123058d01c96d71c8da59527ccc348a7f8ef89.diff) (7973 bytes) - added by adiroiban on 2014-02-27 13:07:21Z - 
* [6994-2.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/47a/47af2040bb2012893930fef26cd9f64525453d04/4513425e2cb556e395ba487f0d7e0c078c4a89cc.diff) (6740 bytes) - added by adiroiban on 2014-04-14 13:56:05Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6994 6994
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__ 
branch__branches_pyflakes_errors_twisted_words_6994 branches/pyflakes-errors-twisted-words-6994
branch_author__cyli cyli
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1393504606000000 1393504606000000
changetime__1397597957000000 1397597957000000
version__None None
owner__ 
cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

I have added a default value for my_address.

While removing unused data-variable from oscar_0E_06 I have also formatted the code as otherwise it was full on errors in my text editor.

Running pydoctor on my computer takes a lot of time and produces a lot of extra errors.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)|@wallrj set owner to @wallrj<br>@wallrj set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_4"></a>|@wallrj set owner to @adiroiban<br>@wallrj set status to `new`|
|-|-|

Thanks Adi,

Notes:
 * Patch applies cleanly
 * No pyflakes errors
 * twisted.words tests all pass locally

Points:
 1. [twisted/words/test/test_domish.py](twisted/words/test/test_domish.py)
    1. "Silence the linter" comment should be removed and use requireModule instead.
    2. Same applies for other such comments in this patch.
 2. source: twisted/words/test/test_jabberxmlstream.py
    1. You're still instantiating a xmlstream.XmlStream object, but what's the point? Why not just remove that line?
    2. Can you also ask the same question for each of the "unused variable" fixes. I guess some of those calls will have side effects, but we check instead of leaving redundant lines of code which may confuse future editors. ie
       ```
       [richard@zorin trunk]$ pyflakes twisted/words | egrep 'but never used'
       twisted/words/protocols/oscar.py:1096: local variable 'data' is assigned to but never used
       twisted/words/protocols/jabber/client.py:278: local variable 'session' is assigned to but never used
       twisted/words/test/test_jabberxmlstream.py:1144: local variable 'xs' is assigned to but never used
       twisted/words/xish/xpathparser.py:366: local variable 'END' is assigned to but never used
       twisted/words/xish/xpathparser.py:383: local variable 'WILDCARD' is assigned to but never used
       twisted/words/xish/xpathparser.py:417: local variable '_context' is assigned to but never used
       twisted/words/xish/xpathparser.py:475: local variable '_context' is assigned to but never used
       twisted/words/xish/xpathparser.py:485: local variable '_context' is assigned to but never used
       ```

Please answer or address the numbered points above and submit a new patch against trunk.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_5"></a>|@ralphm commented|
|-|-|

As mentioned at the top of the file and in the comment removed in this patch, `xpathparser.py` is _generated_ from the `.g` file and must not be altered! Maybe we should exempt this file from being checked by PyFlakes.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban removed owner|
|-|-|


>  1. [twisted/words/test/test_domish.py](twisted/words/test/test_domish.py)
>     1. "Silence the linter" comment should be removed and use
> requireModule instead.

Done


>     2. Same applies for other such comments in this patch.

Done.

>  2. source: twisted/words/test/test_jabberxmlstream.py
>     1. You're still instantiating a xmlstream.XmlStream object, but what's
> the point? Why not just remove that line?

I was thinking that there is some side effect there. Also I wanted to
reduce the scope of this branch to only fix pyflakes errors and not go
into cleaning obscure code.

Line removed.

>    2. Can you also ask the same question for each of the "unused
> variable" fixes. I guess some of those calls will have side effects, but
> we check instead of leaving redundant lines of code which may confuse
> future editors. ie
>        ```
>        [richard@zorin trunk]$ pyflakes twisted/words | egrep 'but never
> used'
>        twisted/words/protocols/oscar.py:1096: local variable 'data' is
> assigned to but never used
>        twisted/words/protocols/jabber/client.py:278: local variable
> 'session' is assigned to but never used
>        twisted/words/test/test_jabberxmlstream.py:1144: local variable
> 'xs' is assigned to but never used
>        twisted/words/xish/xpathparser.py:366: local variable 'END' is
> assigned to but never used
>        twisted/words/xish/xpathparser.py:383: local variable 'WILDCARD' is
> assigned to but never used
>        twisted/words/xish/xpathparser.py:417: local variable '_context' is
> assigned to but never used
>        twisted/words/xish/xpathparser.py:475: local variable '_context' is
> assigned to but never used
>        twisted/words/xish/xpathparser.py:485: local variable '_context' is
> assigned to but never used
>        ```

oscar.py:1096 this line is already removed.

client.py:278 this is a side effect as it adds an element.

I have removed the line from test_jabberxmlstream.py

I have reverted the changes in xpathparser.py.  Ignoring a file is not that
easy as pyflakes has no support for ignoring files.

I think that we will have to create our own pyflakes collector/runner.

---------

I have re-ordered the import in test_msn.py

Added news file.

Please see new patch based on latest trunk.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_7"></a>|@cyli commented|
|-|-|

(In [42294]) Branching to pyflakes-errors-twisted-words-6994.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_8"></a>|@cyli commented|
|-|-|

Thanks for all your hard work, adiroiban!

I've applied the patch - there are no pyflakes errors except on xpathparser.py, which should be exempted.
All twisted tests pass locally.  The buildbot tests look good.

I added extra whitespace to test_msn to make twisted checker pass - when some of the imports were deleted, some whitespace got deleted.

Going to merge.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_9"></a>|@cyli set status to `closed`|
|-|-|

(In [42297]) Merge pyflakes-errors-twisted-words-6994

Author: adiroiban
Reviewer: rwall,ralphm,cyli
Fixes: [#6994](https://github.com/twisted/twisted/issues/6994)
